twine/repository.py

```diff
@@ -77,13 +79,20 @@ def __init__(
 
     @staticmethod
     def _make_adapter_with_retries() -> adapters.HTTPAdapter:
-        retry = urllib3.Retry(
+        retry_kwargs = dict(
             connect=5,
             total=10,
-            method_whitelist=["GET"],
             status_forcelist=[500, 501, 502, 503],
         )
-        return adapters.HTTPAdapter(max_retries=retry)
+
+        # method_whitelist will be removed in 2.0; avoid DeprecationWarning until then
+        urllib3_version = importlib_metadata.version("urllib3")
```

Comment:
Please let's not check the version like this. Why not just do something simple like:

```
try:
    retry = urllib3.Retry(allowed_methods=["GET"], **retry_kwargs)
except TypeError:
    # NOTE remove once urllib3 1.26 is the minimum version
    retry = urllib3.Retry(method_whitelist=["GET"], **retry_kwargs)
```

Reply:
🤦‍♂️ This is much better, thanks. Done.